prov/efa/src/rxr/rxr_read.c

```diff
@@ -509,10 +511,11 @@ int rxr_read_post(struct rxr_ep *ep, struct rxr_read_entry *read_entry)
 			return ret;
 	}
 
-	peer = rxr_ep_get_peer(ep, read_entry->addr);
-
-	if (read_entry->lower_ep_type == SHM_EP)
+	if (read_entry->lower_ep_type == SHM_EP) {
+		peer = rxr_ep_get_peer(ep, read_entry->addr);
```

Comment:
Can you make this change a separate commit because this change actually fixed the bug?

Reply:
Actually, the bug is fixed by handling FI_ADDR_NOTAVAIL in rxr_ep_get_peer. Because we return NULL in time instead of calling the following function `ofi_bufpool_get_ibuf` (was hitting segfault here). 
That's why I made a bit change in my first revision based on our discussion, leaving the call to `rxr_ep_get_peer` outside of the if block.

Reply:
I see. But this change is different from other changes. In other places, we simply add `assert(peer)`, here we moved `rxr_ep_get_peer`. so

Reply:
Yeah. Just not quite sure how to describe this change, as it seems unnecessary, (which is another reason I did not make this change in first revision). Any idea?

Reply:
do not call rxr_ep_get_peer for local read, because local read does not have a peer

Reply:
Thanks, done in new rev.